recipes/protobuf/all/conanfile.py

```diff
@@ -186,15 +220,23 @@ def package(self):
         cmake.install()
         rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
         os.unlink(os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-config-version.cmake"))
-        os.unlink(os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-targets.cmake"))
-        os.unlink(os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-targets-{}.cmake".format(str(self.settings.build_type).lower())))
-        rename(self, os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-config.cmake"),
-                     os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-generate.cmake"))
+        if Version(self.version) < "4.22.0":
+            os.unlink(os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-targets.cmake"))
+            os.unlink(os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-targets-{}.cmake".format(str(self.settings.build_type).lower())))
+            rename(self, os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-config.cmake"),
+                        os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-generate.cmake"))
+        else:
+            # TODO: diary hack, need to fix
+            rename(self, os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-config.cmake"),
+                        os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-protoc.cmake"))
```

Comment:
``` 
          os.unlink(os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-config-version.cmake"))
          os.unlink(os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-targets.cmake"))
          os.unlink(os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-targets-{}.cmake".format(str(self.settings.build_type).lower())))
          if Version(self.version) < "4.22":
              rename(self, os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-config.cmake"),
                           os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-generate.cmake"))
```

Reply:
This hack is only needed for previous versions

Reply:
@miklelappo 
I can't apply it.
In 4.23.1, I have to use protobuf-protoc.cmake(aka protobuf-config.cmake) to define protobuf::protoc target.
And protobuf-config.cmake seems to be not permitted by conan.

Reply:
Got it.. conan magic